li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaFederatedProducerImpl.java

```diff
@@ -62,8 +67,18 @@
     private final Method _boundedFlushMethod;
     private final LongAdder _boundedFlushThreadCount = new LongAdder();
 
-    public PerClusterProducerContext(LiKafkaProducerConfig configs) {
-      _producer = new KafkaProducer<>(configs.originals(), new ByteArraySerializer(), new ByteArraySerializer());
+    public PerClusterProducerContext(ClusterDescriptor cluster, LiKafkaProducerConfig configs) {
+      this(cluster, configs, false);
+    }
+
+    public PerClusterProducerContext(ClusterDescriptor cluster, LiKafkaProducerConfig configs,
+        boolean useMockProducer) {
```

Comment:
i dislike a "mock" flag being used in "prod" code. would it be possible to just provide the MDSClient as an argument ?

Reply:
Added a producer builder as discussed